project/AkkaDependency.scala

```diff
@@ -80,34 +76,23 @@ object AkkaDependency {
                 "com.typesafe.akka" %% module % akkaVersion % config
             },
             resolvers += "Akka library repository".at("https://repo.akka.io/maven"),
-            resolvers ++= (if (akkaSnapshot) Seq("Sonatype Snapshots" at "https://oss.sonatype.org/content/repositories/snapshots") else Nil)
+            resolvers ++= (if (akkaSnapshot) Seq("Akka library snapshot repository".at("https://repo.akka.io/snapshots")) else Nil)
           )
       }
   }
 
-  private def determineLatestSnapshot(prefix: String = ""): String = {
+  private def determineLatestSnapshot(): String = {
     import sbt.librarymanagement.Http.http
     import gigahorse.GigahorseSupport.url
     import scala.concurrent.Await
     import scala.concurrent.duration._
 
-    val snapshotVersionR = """href=".*/((\d+)\.(\d+)\.(\d+)\+(\d+)-[0-9a-f]+-SNAPSHOT)/"""".r
-
-    // akka-cluster-sharding-typed_2.13 seems to be the last nightly published by `akka-publish-nightly` so if that's there then it's likely the rest also made it
-    val body = Await.result(http.run(url("https://oss.sonatype.org/content/repositories/snapshots/com/typesafe/akka/akka-cluster-sharding-typed_2.13/")), 10.seconds).bodyAsString
-    val allVersions =
-      snapshotVersionR.findAllMatchIn(body)
-        .map {
-          case Groups(full, ep, maj, min, offset) =>
-            (
-              ep.toInt,
-              maj.toInt,
-              min.toInt,
-              offset.toInt
-            ) -> full
-        }
-        .filter(_._2.startsWith(prefix))
-        .toVector.sortBy(_._1)
-    allVersions.last._2
+    val versionFile = "https://doc.akka.io/docs/akka/snapshot/paradox.json"
+    val body = Await.result(http.run(url(versionFile)), 10.seconds).bodyAsString
+    val versionProperty = """"version" : """"
+    val i = body.indexOf(versionProperty)
+    if (i == -1)
+      throw new RuntimeException(s"No version in $versionFile")
+    body.substring(i).split('"')(3)
   }
```

Comment:
@johanandren I added the snapshot retrieval. Tried it locally and seems to work, but there is a risk that docs were published but artifacts failed. That is currently the case. Hopefully that will be rare when "that problem" has been solved.

Reply:
I think that's ok